analyzers/src/SonarAnalyzer.Common/AnalysisContext/SonarAnalysisContextBase.cs

```diff
@@ -105,8 +105,18 @@ public bool IsTestProject()
     public bool IsUnchanged(SyntaxTree tree) =>
         UnchangedFilesCache.GetValue(Compilation, _ => CreateUnchangedFilesHashSet()).Contains(tree.FilePath);
 
-    public bool HasMatchingScope(IEnumerable<DiagnosticDescriptor> descriptors) =>
-        descriptors.Any(HasMatchingScope);
+    public bool HasMatchingScope(IEnumerable<DiagnosticDescriptor> descriptors)
+    {
+        // Performance: Don't use descriptors.Any(HasMatchingScope), it allocates too much memory. https://github.com/SonarSource/sonar-dotnet/issues/7438
```

Comment:
💡 If the argument to this is always _already_ an `ImmutableArray<DiagnosticDescriptor>`, it would further help avoid allocations if it wasn't boxed to `IEnumerable<DiagnosticDescriptor>`.

Reply:
The extension method is internal, but the repository is open source. 😄